testutils/infra/device.py

```diff
@@ -153,7 +153,7 @@ def get_reboot_detector(self, host_ip):
     def get_client_service_name(self):
         if self._service_name is None:
             self._service_name = self.run(
-                "if test -e /lib/systemd/system/mender.service; then echo mender; else echo mender-client; fi"
+                "if test -e /lib/systemd/system/mender.service; then echo mender; elif test -e /lib/systemd/system/mender.service; then echo mender; else echo mender-updated; fi"
```

Comment:
```suggestion
                "if test -e /lib/systemd/system/mender.service; then echo mender; elif test -e /lib/systemd/system/mender-client.service; then echo mender-client; else echo mender-updated; fi"
```

But also, what happens to `mender-authd`? For sure some of the tests that use this to restart a service expect both to be restarted :thinking:

Reply:
Can we get the rid of this helper? As far as I recall this was introduced for the transition between yocto releases (when we migrated from `mender` to `mender-client` recipe) but I think this time we will not have the same version of mender client (4.x) with different `systemd` service names, so it should be safe to remove.

@kacf ?

Reply:
I think we cannot. Our service is called mender-updated now, and we decided not to have a mender-client service, so we need to adapt it to that.

Reply:
No, wait, you're right. This only applies to v4.x, so then we don't need it. However, if there are similar constructs in meta-mender, then we do need those, because they will be used with multiple versions of the client.

Reply:
@kacf  Let me insist:

> I think this time we will not have the same version of mender client (4.x) with different systemd service names, so it should be safe to remove

The only scenario where to support multiple systemd service names from integration tests is when one specific mender client release branch (in this case 4.x) can have different systemdservice names based on the yocto branch we are building with.

But as I see the integration going, we either port mender 4.x recipes _and_ the systemd service names to any yocto branch we want to have support for it or none (iow we ~cannot~ will not port 4.x client recipes but keep the old systemd service names for it in an old yocto LTS branch)

Does it make sense?

Reply:
Yes, exactly. Changing the service name is a requirement for any Yocto branch we port it to.

Reply:
Oh, I only now read that you agreed with me in the previous comment. Sorry.

Good :smiley:

Reply:
@oleorhagen You can ignore this for now and either me or you will do on a follow-up PR.